lessons/git/collaboration/lesson.md

```diff
@@ -0,0 +1,495 @@
+all that is needed to get started with contributing to these projects. For
+instance, the source code for the R packages `dplyr` and `ggplot2` is available
+in its entirety on GitHub, and members of the community are welcome to submit
+bug fixes and added features for review by the developers if they so choose.
+Git-based collaboration is more or less the norm in the tech (and scientific
+programming) world, and a valuable skill for anyone doing any sort of
+programming in their work.  
+
+Today, we are going to cover a simple collaborative workflow via Git/GitHub.
+We'll be using relatively simple `bash` operations to navigate and work with
+files throughout: if you need a more detailed refresher on these, feel free to
+refer back to our [Intro Bash][bash-intro] material.
+
+At the start of the lesson, we will be pairing up into pairs of collaborators.
+Although the two collaborators in a pair will still be working through most of
+the lesson individually, there will be steps in which they will take turns to
+see both sides of certain parts of the workflow. Please also make sure to have
+made a [free GitHub account][github] before this lesson. 
+
+# Configuring Git
+
+First-time users will have to do a bit of configuring before getting started
+with Git. If you attended the previous lesson or worked through its material on
+its own time, you will have done this step already, and can skip down to the
+next section. 
+
+First, let's set a username and email:
+
+```bash
+git config --global user.name "Luke Skywalker"
+git config --global user.email "l.skywalker@tattooinet.org"
+```
+
+Git will use `vim` as its default text editor. If you're more inclined to use
+`nano` or `emacs`, the editor can be changed as well:
+
+```bash
+git config --global core.editor nano
+```
+
+Any text editor can be added in here provided you have the respective command
+line utility. For instance, Sublime Text can be added in using `git config
+--global core.editor subl` provided one has `subl` in the `$PATH`.
+
+# Setup
+
+After pairing off, one person in each pair will be designated the 'owner' and
+the other the 'collaborator'. After we have gone through the collaborative
+workflow once, these roles will reverse, so deciding who takes which role first
+is relatively arbitrary. Take a moment now to decide who will be what for
+the first half before we switch sides. 
+
+It is worth noting that many projects (including this workshop series) are
+hosted on a decentralized repo instead of a repo owned by one person. This just
+means that adding to that repo will involve the 'collaborator' workflow instead
+of the 'owner' workflow, while the 'owner' workflow will only apply if one is a
+maintainer/admin of the decentralized repo. 
+
+# Getting started with collaboration - the 'GitHub flow'
+
+Collaboration on GitHub generally centres around the 'GitHub flow'. This makes
+use of features we've seen before, such as forks and branches, but have yet to
```

Comment:
They will not yet have heard of 'forks and branches', right?

Reply:
I meant more that they may have seen the terms last lesson (a few people asked what 'on branch master' was) but we didn't do anything with them. I've restated this to make it clearer